### PR TITLE
hyprsome: init at 0.1.12

### PR DESCRIPTION
**Pull request by BhasherBEL**
Adds [hyprsome](https://github.com/sopa0/hyprsome), a tool for interacting with hyprland workspaces. Up to date version of #259495 which is now stale for long.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by zi3m5f — COMMENTED** [review]
Their own [README.md](https://github.com/sopa0/hyprsome/blob/master/README.md#alternatives) says:
> ### Alternatives
> https://github.com/Duckonaut/split-monitor-workspaces leverages the plugin system, is more feature rich and seems better maintained

And [split-monitor-workspaces](https://github.com/Duckonaut/split-monitor-workspaces#special-thanks):
> [hyprsome](https://github.com/sopa0/hyprsome): An earlier project of similar nature

Sure you don't want to add the other one? ;)

How can I check version 0.1.12 is `rev = "985e1a3b03b5118c98c03983f60ea9f74775858c";`?

You missed https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
> When adding yourself as maintainer in the same pull request, make a separate commit with the message maintainers: add <handle>. Add the commit before those making changes to the package or module. See Nixpkgs Maintainers for details.

**Comment by BhasherBEL** [comment]
> Sure you don't want to add the other one? ;)

They indeed cite another tool, but their is still active and used :) 
 
> How can I check version 0.1.12 is `rev = "985e1a3b03b5118c98c03983f60ea9f74775858c";`?

Sadly they don't have special branches, tags, ... for versions. It just _happens_ that the last version is `0.1.12`.

The other points you raised should be fixed. Thanks!

**Review by zi3m5f — CHANGES_REQUESTED** [review]
### Reviewed points

- [x] package path fits guidelines
- [x] package name fits guidelines
- [ ] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [ ] `meta.mainProgram` is set, if applicable.
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

### Possible improvements

> Sadly they don't have special branches, tags, ... for versions. It just happens that the last version is 0.1.12.

According to [pkgs/README.md#versioning](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning):
> If a package is a commit from a repository without a version assigned, then the version attribute should be the latest upstream version preceding that commit, followed by -unstable- and the date of the (fetched) commit. The date must be in "YYYY-MM-DD" format.

And seeing the version got bumped to 0.1.12 in an earlier [commit](https://github.com/sopa0/hyprsome/commit/c77b7d4b5309f4c0edcaea344ddb29cc947d3c48) at 2024-05-02 and you chose a later [commit](https://github.com/sopa0/hyprsome/commit/985e1a3b03b5118c98c03983f60ea9f74775858c) from 2024-05-20 the correct version would be `0.1.12-unstable-2024-05-20`.

Other than that it looks good:

### `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352677`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hyprsome</li>
  </ul>
</details>

### Nice to have

Not required but maybe you could add [`versionCheckHook`](https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook)?

If you do, this avoids calling with "--help" first:
```nix
  versionCheckProgramArg = "--version";
```

**Review by zi3m5f — CHANGES_REQUESTED** [review]
Minus `meta.versionCheckProgramArg` it LGTM :tada:

**Review by zi3m5f — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352677`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>hyprsome</li>
  </ul>
</details>

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/hy/hyprsome/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "985e1a3b03b5118c98c03983f60ea9f74775858c";
+    hash = "sha256-JiOV9c23yOhaVW2NHgs8rjM8l9qt181Tigf5sCnPep8=";
+  };
+
+  cargoHash = "sha256-Jqx77/PDWEJ+NPJN2R6n4w9nVPXQoHeuVfYKUB3i9Lg=";
+
+  meta = {
+    description = "Awesome-like workspaces for Hyprland";
+    homepage = "https://github.com/sopa0/hyprsome";
+    license = lib.licenses.gpl3;
```
Comment (Aleksanaa) [inline pkgs/by-name/hy/hyprsome/package.nix:23]:
```suggestion
    license = lib.licenses.gpl3Only;
```

`lib.licenses.gpl3` is deprecated.

**Pull request merged**